lib/segment/src/index/field_index/numeric_index/mod.rs

```diff
@@ -526,21 +522,27 @@ impl ValueIndexer<FloatPayloadType> for NumericIndex<FloatPayloadType> {
         NumericIndex::remove_point(self, id)
     }
 }
+impl<T: Numericable> From<RangeInterface> for Range<T> {
+    fn from(value: RangeInterface) -> Self {
+        match value {
+            RangeInterface::Float(float_range) => float_range.map(T::from_f64),
+            RangeInterface::Int(int_range) => int_range.map(T::from_i64),
+            RangeInterface::DateTime(datetime_range) => {
+                datetime_range.map(|dt| T::from_i64(dt.timestamp()))
+            }
+        }
+    }
+}
 
 impl<T> StreamRange<T> for NumericIndex<T>
 where
     T: Encodable + Numericable + Default,
 {
     fn stream_range(
         &self,
-        range: &RangeInterface,
+        range_interface: &RangeInterface,
     ) -> Box<dyn DoubleEndedIterator<Item = (T, PointOffsetType)> + '_> {
-        let range = match range {
-            RangeInterface::Float(float_range) => float_range.map(T::from_f64),
-            RangeInterface::DateTime(datetime_range) => {
-                datetime_range.map(|dt| T::from_i64(dt.timestamp()))
-            }
-        };
+        let range = Range::from(range_interface.clone());
```

Comment:
Nit: maybe we could avoid cloning if there is `impl<T: Numericable> From<&RangeInterface> for Range<T>` (note the `&`).

Reply:
Yeah, I also thought about that, but I think this is a fairly inexpensive operation.

To make this more explicit, I have just pushed a change to make `Range` and `RangeInterface` `Copy`